src/main/java/org/cloudfoundry/promregator/auth/OAuth2XSUAAEnricher.java

```diff
@@ -1,181 +1,103 @@
-		httpPost.setHeader("Content-Type", "application/json");
-
-		/* closing the connection afterwards is important!
-		 * Background: httpclient will otherwise try to keep the connection open.
-		 * We won't be calling often anyway, so the server would be drained from resources.
-		 * Moreover, if the server has gone away in the meantime, the next attempt to 
-		 * call would fail with a recv error when reading from the socket.
-		 */
-		httpPost.setHeader("Connection", "close");
-		
-		List<NameValuePair> form = new ArrayList<>();
-		form.add(new BasicNameValuePair("response_type", "token"));
-		UrlEncodedFormEntity formEntity = new UrlEncodedFormEntity(form, Consts.UTF_8);
-		httpPost.setEntity(formEntity);
-		
-		CloseableHttpResponse response = null;
-		String json = null;
+
+		log.info("Fetching new JWT token");
+
+		TokenResponse oAuthResponse = null;
 		try {
-			try {
-				response = httpclient.execute(httpPost);
-			} catch (ClientProtocolException e) {
-				log.error("Unable to read from the token server", e);
-				return null;
-			} catch (IOException e) {
-				log.error("IO Exception while reading from the token server", e);
-				return null;
-			}
-			
-			if (response.getStatusLine().getStatusCode() != 200) {
-				log.error(String.format("Server did not respond with ok while fetching JWT from token server; status code provided: %d", response.getStatusLine().getStatusCode()));
-				return null;
-			}
-			
-			try {
-				json = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
-			} catch (ParseException e) {
-				log.error("GSON parser exception on JWT response from token server", e);
-				return null;
-			} catch (IOException e) {
-				log.error("IO Exception while running GSON parser", e);
+			oAuthResponse = tokenFetcher.getJWT(config);
+			if (oAuthResponse == null) {
+				log.error("Cannot retrieve token");
 				return null;
 			}
-		} finally {
-			if (response != null) {
-				try {
-					response.close();
-				} catch (IOException e) {
-					log.info("Unable to properly close JWT-retrieval connection", e);
-				}
-			}
-		}
-		
-		if (json == null) {
-			log.warn("Null-JSON detected on OAuth response");
+		} catch (IOException e) {
+			log.error("Cannot retrieve token", e);
 			return null;
 		}
-		
-		Gson gson = new Gson();
-		TokenResponse oAuthResponse = gson.fromJson(JsonSanitizer.sanitize(json), TokenResponse.class);
-		
+
 		String jwt = oAuthResponse.getAccessToken();
 		log.info(String.format("JWT token retrieved: %s...", jwt.substring(0, Math.min(jwt.length() / 2, 30))));
 
 		int timeOutForUs = Math.max(oAuthResponse.getExpiresIn() - 30, oAuthResponse.getExpiresIn() / 2);
 		this.validUntil = Instant.now().plus(timeOutForUs, ChronoUnit.SECONDS);
-		
+
 		log.info(String.format("JWT is valid until %s", this.validUntil.toString()));
-		
+
 		return jwt;
 	}
 
-	private static class TokenResponse {
+	static class TokenResponse {
```

Comment:
If we use it outside, we also should make it more prominent.
We may even consider having a package like `org.cloudfoundry.promregator.auth.xsuaa`  or `org.cloudfoundry.promregator.auth.jwt` for that.